File: Algorithm_119_Pascal's Triangle II.cpp
#include <iostream>
#include <vector>
using namespace std;
vector<int> getRow(int rowIndex)
{
	vector<int> tmp;
	tmp.push_back(1);
	if (rowIndex==0) return tmp;
	for (int i=1;i<=rowIndex;i++)
	{
		vector<int> last(tmp.begin(),tmp.end());
		tmp.clear();
		for (int j=0;j<=i;j++)
		{
			int left=(j==0)?0:last[j-1];
			int right=(j==i)?0:last[j];
			tmp.push_back(left+right); 
		}
	}
	return tmp;
}

File: Algorithm_119_Pascal's Triangle II.py
class Solution:
    def getRow(self, rowIndex: int) -> List[int]:
        result = [0 for _ in range(rowIndex + 1)]
        result[0] = 1
        for i in range(1, rowIndex + 1):
            for j in range(i, 0, -1):
                result[j] += result[j - 1]
        return result